msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/ManagedIdentityResponse.java

```diff
@@ -3,42 +3,67 @@
 
 package com.microsoft.aad.msal4j;
 
-import com.fasterxml.jackson.annotation.JsonCreator;
-import com.fasterxml.jackson.annotation.JsonProperty;
+import com.azure.json.JsonReader;
+import com.azure.json.JsonSerializable;
+import com.azure.json.JsonToken;
+import com.azure.json.JsonWriter;
 import lombok.Getter;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.io.IOException;
+
 @Getter
-class ManagedIdentityResponse {
+class ManagedIdentityResponse implements JsonSerializable<ManagedIdentityResponse> {
 
     private static final Logger LOG = LoggerFactory.getLogger(ManagedIdentityResponse.class);
 
-    @JsonProperty(value = "token_type")
     String tokenType;
-
-    @JsonProperty(value = "access_token")
     String accessToken;
-
-    @JsonProperty(value = "expires_on")
     String expiresOn;
-
     String resource;
-
-    @JsonProperty(value = "client_id")
     String clientId;
 
-    /**
-     * Creates an access token instance.
-     *
-     * @param token the token string.
-     * @param expiresOn the expiration time.
-     */
-    @JsonCreator
-    private ManagedIdentityResponse(
-            @JsonProperty(value = "access_token") String token,
-            @JsonProperty(value = "expires_on") String expiresOn) {
-        this.accessToken = token;
-        this.expiresOn =  expiresOn;
+    public static ManagedIdentityResponse fromJson(JsonReader jsonReader) throws IOException {
```

Comment:
Jackson seems more elegant / readable with the attribute approach...

Reply:
Jackson Databind is much cleaner, and in some cases clearer, but it does require a lot of reflection to be used by Jackson. This code here is effectively equivalent to what you'd see if you were using Jackson Core which doesn't have all the reflection handling.